src/map.rs

```diff
@@ -88,6 +91,44 @@ pub struct IndexMap<K, V, S> {
     hash_builder: S,
 }
 
+#[cfg(feature = "merge")]
+impl<K, V, S> Merge for IndexMap<K, V, S>
+where
+    K: Hash + Eq,
+    S: BuildHasher,
+{
+    /// Note: If any of the rows have the same key, then they wont be merged.
+    /// the value on the right will be preserved.
```

Comment:
This is contradicted by the example, where you show that `a.merge(b)` preserves the left values.

But either way *could* be desired, so I don't think it makes sense for us to choose. Users can derive `Merge` with their own annotation to deal with `IndexMap` in the way that works best for them.

Reply:
I will fix the typo in that bit of the comment, in regards to order, the merge crate does actually set a precedent, namely in the options section, it specifies that the left value should be used.

From the test section of the merge crate:
```rust
fn test<T: std::fmt::Debug + Merge + PartialEq>(expected: T, mut left: T, right: T) {
    left.merge(right);
    assert_eq!(expected, left);
}

#[derive(Debug, Merge, PartialEq)]
struct S {
    field1: Option<usize>,
}

impl S {
    pub fn new(field1: Option<usize>) -> S {
        S { field1 }
    }
}

test(S::new(Some(1)), S::new(Some(1)), S::new(Some(2)));
```

When specialization lands in the rust compiler, the user will even be able to overwrite it with their own implementation. It's just something that most people will do in the same way, so it makes sense to implement it for everyone.

Reply:
Ok, but still, `Option` is the *only* provided implementation. As I mentioned in #286, they did not choose the same for `Vec` or `HashMap`, which would be more directly applicable to `IndexMap`.

You don't need specialization -- just derive your type with a custom `#[merge(strategy = "your_func")]` on your fields.

Reply:
You... Make a good point :(
This is the first time I've tried to contribute to an open source project, I guess I got a bit ahead of myself. Sorry for the inconvenience.

Reply:
It's okay! It may not feel like it, but the discussion is a contribution as well, so please don't let this particular outcome deter you.